src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueCreateDialog.kt

```diff
@@ -82,22 +84,46 @@ class IssueCreateDialog(
             openConfigureFieldsDialog(dialogLocator)
         }
 
-        val allFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
-        if (driver.isElementPresent(allFieldsLinkLocator)) {
-            driver.wait(elementToBeClickable(allFieldsLinkLocator)).click()
-            val newAllFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
-            driver.wait(not(elementToBeClickable(newAllFieldsLinkLocator)))
+        if (newConfigureFieldsUi) {
+            val selectLocator = getFieldsSelectLocator(configureFieldsDialogClass);
+
+            if (driver.isElementPresent(selectLocator)) {
+                val s = Select(driver.findElement(selectLocator))
+
+                if (s.firstSelectedOption.text != "All Fields") {
+                    s.selectByVisibleText("All Fields")
+
+                    driver.wait(elementToBeClickable(selectLocator));
+                }
+            }
+        } else {
+            val allFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
+            if (driver.isElementPresent(allFieldsLinkLocator)) {
+                driver.wait(elementToBeClickable(allFieldsLinkLocator)).click()
+                val newAllFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
+                driver.wait(not(elementToBeClickable(newAllFieldsLinkLocator)))
+            }
         }
 
         dismissConfigureFieldsDialog()
         return this
     }
 
+    private fun isNewConfigureFieldsUiAvailable(): Boolean {
+        return driver.findElement(By.cssSelector("meta[name=ajs-build-number]"))
+            .getAttribute("content")
+            .toInt() >= 819000
```

Comment:
Does the new AJS provide semantic DOM to rely on?

Reply:
What do you mean by a new AJS? Meta tags are rendered within <head> and are in Jira since... ages?

Reply:
Sorry, that's my Jira frontend ignorance speaking :sweat_smile: 
I meant: does Jira 8.19 provide semantic (long-term) selectors for new GUI? it's a practice similar to feature detection vs browser detection

Reply:
For the Configure Fields it does - there is a unique identifier and some data-test-id attributes, anyway, as per another comment I'll base on feature detection and on the presence of the new element

Reply:
The other thread is about 8.14 stuff. If 8.14 selectors are sufficient for 8.19, then this convo can be resolved.

Reply:
So we got 8.19 compat without version detection, noice!